packages/preset-rules/src/customRules.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as fs from 'fs';
+import path from 'path';
+import axe from 'axe-core';
+
+// Directory containing custom rules and checks
+const customRulesPathDir = 'custom-rules';
+
+export function registerCustomRules(): void {
+    const directoryPath = path.join(__dirname, customRulesPathDir);
+    const newChecks: axe.Check[] = [];
+    const newRules: axe.Rule[] = [];
+
+    try {
+        // Path to the JSON file with changes to existing rules
+        const changesInExistingRulesPath = path.join(directoryPath, 'changes.json');
+        // Read and parse existing rule changes
+        const existingRuleData = fs.readFileSync(changesInExistingRulesPath, 'utf-8');
+        const { rules } = JSON.parse(existingRuleData) as { rules: axe.Rule[] };
+        if (rules && Array.isArray(rules)) {
+            newRules.push(...rules);
+        }
+
+        // Function to process JSON files in a directory and push their content to a target array
+        const processJsonFiles = <T>(dir: string, targetArray: T[]): void => {
```

Comment:
for the sake of cleaner and effective coding, I would suggest to push this utility code under `@sa11y/common` more generic to accept type of file extension too